### PR TITLE
Add refresh token before expire

### PR DESCRIPTION
**Pull request by khalifapro**

### CODE REVIEW

File: auth0/token.go [hunk 1]
```diff
@@ -80,32 +80,31 @@ func (a *ClientProvider) GetToken() (string, error) {
...
 		return authToken, err
 	}
+
 	// check token validity
-	ok, err := a.isValid(authToken)
+	ok, claims, err := a.isValid(authToken)
 	if ok {
+		go func() {
+			// refresh token before expiry by X minutes
+			if claims.VerifyExpiresAt(time.Now().Add(5*time.Minute).Unix(), false) == true {
```
Comment (lukaszgryglicki) [inline auth0/token.go:96]: I would do just 1 minute instead of 5.

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm

File: auth0/token.go (hunk 1 above)
Comment (khalifapro) [reply auth0/token.go:96]: it was Fayaz request

Comment (lukaszgryglicki) [reply auth0/token.go:96]: yes but 5/60 is almost 10% of the token lifetime, while the problem probably happens when the library returns a still valid token and then we attempt to use it (so this is a few seconds at most). I would prefer 1 minute...

**Pull request merged**